testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/elytron/CustomCredentialSecurityFactoryTestCase.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.manualmode.elytron;
+
+import static org.apache.http.HttpStatus.SC_OK;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.http.HttpStatus;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.security.common.CoreUtils;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ * Test for authentication through http-interface secured by Elytron http-authentication-factory where we test that
+ * custom-credential-security-factory is called.
+ *
+ * @author olukas
+ * @author Hynek Švábek <hsvabek@redhat.com>
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class CustomCredentialSecurityFactoryTestCase {
+
+    private static final String PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY = "global";
+    private static final String MANAGEMENT_FILESYSTEM_NAME = "mgmt-filesystem-name";
+    private static final String CUSTOM_CREDENTIAL_SECURITY_FACTORY_MODULE_NAME = "org.jboss.customcredentialsecurityfactoryimpl";
+    private static final String CUSTOM_CRED_SEC_FACTORY_NAME = "customCredSecFactory";
+
+    private static File tempFolder;
+
+    private static final String USER = "user";
+    private static final String CORRECT_PASSWORD = "password";
+
+    private static String host;
+
+    @Inject
+    private static ServerController CONTROLLER;
+
+    public static void prepareServerConfiguration() throws Exception {
+        tempFolder = File.createTempFile("ely-" + CustomCredentialSecurityFactoryTestCase.class.getSimpleName(), "", null);
+        if (tempFolder.exists()) {
+            tempFolder.delete();
+            tempFolder.mkdir();
+        }
+
+        String fsRealmPath = tempFolder.getAbsolutePath() + File.separator + "fs-realm-users";
```

Comment:
Just as an example if you change the `tempFolder` to a `java.nio.file.Path` like suggested above this would just become:
```java
String fsRealmPath = tempFolder.resolve("fs-realm-users").toString();
```